### ISSUE TITLE
--auto-gen-config fails to add exclude some failures in .rubocop_todo.yml

### ISSUE THREAD

**Issue by ClayShentrup** [bbatsov/rubocop#3340]
Running `--auto-gen-config` on this method adds a Style/GuardClause exclusion to .rubocop_todo.yml.

```ruby
def foobar
  if true
    return 'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefg'
  end
end
```

But add one extra character to that string and you get this:

```ruby
def foobar
  if true
    return 'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefgh'
  end
end
```

Which DOES NOT add a Style/GuardClause exclusion to the todo file. _But it still fails_.

```bash
filepath:213:5: C: Use a guard clause instead of wrapping the code inside a conditional expression.
    if true
    ^^
```

--------

## Expected behavior
Running with `--auto-gen-config` adds a Style/GuardClause exclusion to `.rubocop_todo.yml`, so that running `rubocop` after running `rubocop --auto-gen-config` will always pass.

## Actual behavior
Some failures aren't added to `.rubocop_todo.yml`, meaning that this command can fail:

```bash
rubocop --auto-gen-config ; rubocop
```

## Steps to reproduce the problem
Run `rubocop --auto-gen-config` on the `foobar` methods described above.

## RuboCop version

0.42.0
ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin15]

**Comment by jonas054** [comment]
Closing as a duplicate of #2679.

**Comment by ClayShentrup** [comment]
@jonas054 I don't think this is a duplicate of 2679. That bug is about a to-do rule being ignored. This bug is about the line not even getting added to the to-do list even though it fails. If I added it manually, it's respected perfectly fine.

**Comment by jonas054** [comment]
OK, let's reopen and continue the discussion. I still think it's the same root cause, but we'll see if that's the case or not.

First of all, I can't reproduce your problem. Try creating an empty directory with just one file containing the `foobar` method, and see if you can reproduce. Maybe you have a setting for `Metrics/LineLength: Max` somewhere.

**Comment by ClayShentrup** [comment]
@jonas054 Okay, I'll do some more investigation. But I don't see how a `Metrics/LineLength` or any other setting could be relevant. Again, the failure happens when running rubocop, but it just doesn't get added to the todolist file.

**Comment by ClayShentrup** [comment]
Also I don't see how you would expect the same root cause when the problems seem to me to be completely different. My problem is that the line doesn't get added to the todolist. But _if_ I manually add it, it is respected. #2679 is about a rule that _is_ in the todolist not being respected.

**Comment by ClayShentrup** [comment]
This is interesting so far. When running with `--auto-gen-config`, I get the results of [this line](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/runner.rb#L103).

```ruby
offenses = yield
```

The file in question, which has a GuardClause exception, _has no record_ for `RuboCop::Cop::Offense`! It has a host of others.

But if I run it without `--auto-gen-config`, boom, I get this.

```ruby
#<RuboCop::Cop::Offense:0x007fe6626a6490
  @cop_name="Style/GuardClause",
  @location=#<Parser::Source::Range FILEPATH>,
  @message="Use a guard clause instead of wrapping the code inside a conditional expression.",
  @severity=#<RuboCop::Cop::Severity:0x007fe6626a6468 @name=:convention>,
  @status=:unsupported>
```

Continuing to investigate.

**Comment by ClayShentrup** [comment]
This gets even more interesting. I have a file with a GuardClause issue. The issue is ignored unless I add this to my todo file:

```
Metrics/LineLength:
  Max: 202
```

Strangely, the longest line in the file is 98 characters long.

**Comment by ClayShentrup** [comment]
This line looks like the culprit.

```ruby
indent + (source + keyword + condition.source).length + 2 > max
```

That precisely equals the above max length.

**Comment by ClayShentrup** [comment]
So basically this logic is saying, don't fail for this Style/GuardClause cop if autocorrecting it would make the line too long. But that should only determine whether to autocorrect it, not whether to fail for it.

**Comment by ClayShentrup** [comment]
@jonas054 I now see how this is a duplicate. Thanks.
